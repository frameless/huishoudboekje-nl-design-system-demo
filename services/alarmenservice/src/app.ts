import cors from "cors";
import express from "express";
import apiV1Router from "./api-v1";
import {errorHandler} from "./errorHandlers";
import healthRouter from "./health";

const app = express();
const DATABASE_URL = process.env.DATABASE_URL;

if (!DATABASE_URL && process.env.NODE_ENV !== "test") {
	throw new Error("Unknown database. Please provide the database connection string in DATABASE_URL in the environment.");
}

const logger = () => (req, res, next) => {
	console.log("[EVENT]", `Incoming request: ${req.method} ${req.url}`);
	if (req.body) {
		console.log("[EVENT]", "Body:", req.body);
	}
	next();
};

app.use(cors());
app.use(express.json());
if (process.env.NODE_ENV === 'dev') {
	app.use(logger());
}

app.get("/health", healthRouter);
app.use("/v1/alarms", apiV1Router);
app.use(errorHandler);

export default app;