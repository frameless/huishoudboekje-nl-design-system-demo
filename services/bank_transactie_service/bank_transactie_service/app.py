""" Main app for bank_transactie_service """
import logging
import logging.config

from flask import Flask, Response
from flask_migrate import Migrate

from bank_transactie_service.views.bank_transaction import BankTransactionView
from bank_transactie_service.views.customer_statement_message import CustomerStatementMessageView
from core_service import database
from bank_transactie_service.views.bank_transaction_range import BanktransactionRangeView
from bank_transactie_service.views.transactions_filter_view import BanktransactionFilterView
from bank_transactie_service.views.bank_transaction_sum import BanktransactionSumView
from core_service.sqlalchemy_statsd_metrics import add_sqlalchemy_statsd_metrics

db = database.db


def create_app(config_name='bank_transactie_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    db.init_app(app)
    Migrate(app, db)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    # logging.config.dictConfig(
    #     {
    #         "version": 1,
    #         "incremental": True,
    #         "loggers": {"sqlalchemy.engine": {"level": app.config["LOG_LEVEL"]}},
    #     }
    # )
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel(app.config["LOG_LEVEL"])

    add_sqlalchemy_statsd_metrics(app)


    @app.route('/health')
    def health():
        return Response()

    # Views
    routes = [
        {"path": "/customerstatementmessages", "view": CustomerStatementMessageView,
            "name": "customer_statement_message_view"},
        {"path": "/customerstatementmessages/<object_id>", "view": CustomerStatementMessageView,
            "name": "customer_statement_message_detail_view"},
        {"path": "/banktransactions", "view": BankTransactionView,
            "name": "banktransaction_view"},
        {"path": "/banktransactions/<object_id>", "view": BankTransactionView,
            "name": "banktransaction_detail_view"},
        {"path": "/banktransactions/filter", "view": BanktransactionFilterView,
            "name": "banktransaction_filter_view"},
        {"path": "/banktransactions/range", "view": BanktransactionRangeView,
            "name": "banktransaction_range_view"},
        {"path": "/banktransactions/sum", "view": BanktransactionSumView,
            "name": "banktransaction_sum_view"},
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app


if __name__ == '__main__':
    create_app().run()
