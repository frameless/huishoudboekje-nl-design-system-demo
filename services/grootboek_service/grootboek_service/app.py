""" Main app for grootboek_service """
from models import *
import logging
from flask import Flask, Response
from flask_migrate import Migrate
from core_service import database
from grootboek_service.views.grootboekrekeningen import GrootboekrekeningenView
from core_service.statsd_metrics import add_statsd_metrics

db = database.db


def create_app(config_name='grootboek_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel(app.config["LOG_LEVEL"])

    db.init_app(app)
    Migrate(app, db)

    add_statsd_metrics(app)

    @app.route('/health')
    def health():
        return Response()

    # Views
    routes = [
        {"path": "/grootboekrekeningen", "view": GrootboekrekeningenView,
         "name": "grootboekrekening_view"},
        {"path": "/grootboekrekeningen/<object_id>", "view": GrootboekrekeningenView,
         "name": "grootboekrekening_detail_view"}
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app


if __name__ == '__main__':
    create_app().run()
