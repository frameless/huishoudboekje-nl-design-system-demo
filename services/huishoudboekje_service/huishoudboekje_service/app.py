""" Main app for huishoudboekje_service """
import logging
import os
from flask import Flask, Response
from flask_migrate import Migrate

from huishoudboekje_service.views import (
    BurgerView,
    AfspraakView,
    RekeningView,
    RekeningBurgerView,
    RekeningAfdelingView,
    JournaalpostView,
    RubriekView,
    ConfiguratieView,
    OverschrijvingView,
    ExportView,
    HuishoudenView,
    AfdelingView,
)
from core_service import database

db = database.db


def create_app(config_name='huishoudboekje_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    logging.info(f"Starting {__name__} with {config_name}")

    db.init_app(app)
    migrate = Migrate(app, db)

    @app.route('/health')
    def health():
        return Response()

    # Views
    routes = [
        {"path": "/burgers", "view": BurgerView, "name": "burger_view"},
        {"path": "/burgers/<object_id>", "view": BurgerView, "name": "burger_detail_view"},
        {"path": "/burgers/<object_id>/rekeningen", "view": RekeningBurgerView,
         "name": "burger_rekeningen_view"},
        {"path": "/afspraken", "view": AfspraakView, "name": "afspraak_view"},
        {"path": "/afspraken/<object_id>", "view": AfspraakView, "name": "afspraak_detail_view"},
        {"path": "/rekeningen", "view": RekeningView, "name": "rekening_view"},
        {"path": "/rekeningen/<object_id>", "view": RekeningView, "name": "rekening_detail_view"},
        {"path": "/journaalposten", "view": JournaalpostView, "name": "journaalpost_view"},
        {"path": "/journaalposten/<object_id>", "view": JournaalpostView, "name": "journaalpost_detail_view"},
        {"path": "/rubrieken", "view": RubriekView, "name": "rubriek_view"},
        {"path": "/rubrieken/<object_id>", "view": RubriekView, "name": "rubriek_detail_view"},
        {"path": "/configuratie", "view": ConfiguratieView, "name": "configuratie_view"},
        {"path": "/configuratie/<object_id>", "view": ConfiguratieView, "name": "configuratie_detail_view"},
        {"path": "/overschrijvingen", "view": OverschrijvingView, "name": "overschrijving_view"},
        {"path": "/overschrijvingen/<object_id>", "view": OverschrijvingView, "name": "overschrijving_detail_view"},
        {"path": "/export", "view": ExportView, "name": "export_view"},
        {"path": "/export/<object_id>", "view": ExportView, "name": "export_detail_view"},
        {"path": "/huishoudens", "view": HuishoudenView, "name": "huishouden_view"},
        {"path": "/huishoudens/<object_id>", "view": HuishoudenView, "name": "huishouden_detail_view"},
        {"path": "/afdelingen", "view": AfdelingView, "name": "afdeling_view"},
        {"path": "/afdelingen/<object_id>", "view": AfdelingView, "name": "afdeling_detail_view"},
        {"path": "/afdelingen/<object_id>/rekeningen", "view": RekeningAfdelingView,
         "name": "afdeling_rekeningen_view"},
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app
