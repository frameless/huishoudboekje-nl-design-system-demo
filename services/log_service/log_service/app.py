""" Main app for log_service """
import logging
import logging.config

from flask import Flask, Response
from flask_migrate import Migrate

from core_service import database
from log_service.views.gebruikersactiviteit import GebruikersActiviteitView
from core_service.statsd_metrics import add_statsd_metrics

db = database.db


def create_app(
    config_name="log_service.config.Config"
):
    app = Flask(__name__)
    app.config.from_object(config_name)

    db.init_app(app)
    Migrate(app, db)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    # logging.config.dictConfig(
    #     {
    #         "version": 1,
    #         "incremental": True,
    #         "loggers": {"sqlalchemy.engine": {"level": app.config["LOG_LEVEL"]}},
    #     }
    # )
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel("WARNING")

    add_statsd_metrics(app)

    @app.route("/health")
    def health():
        return Response()

    # Views
    routes = [
        {
            "path": "/gebruikersactiviteiten",
            "view": GebruikersActiviteitView,
            "name": "gebruikersactiviteit_view",
        },
        {
            "path": "/gebruikersactiviteiten/<object_id>",
            "view": GebruikersActiviteitView,
            "name": "gebruikersactiviteit_detail_view",
        },
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False,
        )

    return app


if __name__ == "__main__":
    create_app().run()
