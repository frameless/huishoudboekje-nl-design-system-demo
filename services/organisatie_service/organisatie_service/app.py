""" Main app for organisatie_service """
from models import *
import logging

from flask import Flask, Response
from flask_migrate import Migrate

from organisatie_service.views.organisaties import OrganisatieView
from organisatie_service.views.organisatie_rekeningen import OrganisatieRekeningenView
from organisatie_service.views.afdelingen import AfdelingView
from core_service import database
from core_service.sqlalchemy_statsd_metrics import add_sqlalchemy_statsd_metrics

db = database.db


def create_app(config_name='organisatie_service.config.Config'):
    app = Flask(__name__)
    app.config.from_object(config_name)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel(app.config["LOG_LEVEL"])

    db.init_app(app)
    Migrate(app, db)

    add_sqlalchemy_statsd_metrics(app)

    @app.route('/health')
    def health():
        return Response()

    routes = [
        {"path": "/organisaties", "view": OrganisatieView, "name": "organisatie_view"},
        {"path": "/organisaties/<object_id>", "view": OrganisatieView,
            "name": "organisatie_detail_view"},
        {"path": "/organisaties/rekeningen", "view": OrganisatieRekeningenView, "name": "organisatie_rekeningen_view"},
        {"path": "/afdelingen", "view": AfdelingView, "name": "afdeling_view"},
        {"path": "/afdelingen/<object_id>", "view": AfdelingView,
            "name": "afdeling_detail_view"},
    ]

    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )
    return app


if __name__ == '__main__':
    create_app().run()
