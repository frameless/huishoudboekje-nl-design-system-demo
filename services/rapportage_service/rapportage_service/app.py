""" Main app for rapportage_service """
import logging
import logging.config

from flask import Flask, Response
from flask_injector import FlaskInjector
from rapportage_service.dependencies import configure
from rapportage_service.views.RapportageView import BurgerRapportageView


def create_app(config_name='rapportage_service.config.Config', dependency_injection_configuration=configure):
    app = Flask(__name__)
    app.config.from_object(config_name)

    logging.basicConfig(
        format='%(asctime)s %(levelname)-8s %(message)s',
        level=app.config["LOG_LEVEL"],
        datefmt='%Y-%m-%d %H:%M:%S')
    logging.config.dictConfig(
        {
            "version": 1,
            "incremental": True,
            "loggers": {"sqlalchemy.engine": {"level": app.config["LOG_LEVEL"]}},
        }
    )
    logging.info(f"Starting {__name__} with {config_name}")

    # Werkzeug has their own logger which outputs info level URL calls.
    # This can also cause parameters that are normally hidden to be logged
    logging.getLogger('werkzeug').setLevel(app.config["LOG_LEVEL"])

    @app.route('/health')
    def health():
        return Response()

    # Views
    routes = [
        {"path": "/rapportage", "view": BurgerRapportageView,
         "name": "rapportage_burger"}
    ]
    for route in routes:
        app.add_url_rule(
            route["path"],
            view_func=route["view"].as_view(route["name"]),
            strict_slashes=False
        )

    # Initialize Flask-Injector. This needs to be run after you attached all
    # views, handlers, context processors and template globals.
    FlaskInjector(app=app, modules=[dependency_injection_configuration])
    return app


if __name__ == '__main__':
    create_app().run()
