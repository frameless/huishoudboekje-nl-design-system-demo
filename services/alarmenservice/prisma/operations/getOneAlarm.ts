import prisma from "../../src/client";
import {NotFoundError} from "../../src/errorHandlers";

const getOneAlarm = async (id: string) => {
	const data = await prisma.alarm.findUnique({
		where: {id},
	});

	if (!data) {
		throw new NotFoundError();
	}

	return data;
};

export default getOneAlarm;