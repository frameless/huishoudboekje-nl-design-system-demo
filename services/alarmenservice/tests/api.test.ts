import {Alarm, DayOfWeek, Prisma} from "@prisma/client";
import supertest from "supertest";
import {prismaMock} from "../.jest/mockClient";
import getOneAlarm from "../prisma/operations/getOneAlarm";
import app from "../src/app";
import alarms from "./fixtures/alarms.json";
import ResolvedValue = jest.ResolvedValue;

const api = supertest(app);

const toAlarm = (data: any): Alarm => {
	return {
		...data,
		byDay: alarms[0]!.byDay.map(a => {
			return DayOfWeek[a];
		}),
	};
};

describe("Alarm CRUD (operations)", () => {

	describe("Given the Alarm exists", () => {
		it("should return the Alarm", async () => {
			const alarm = toAlarm(alarms[0]);
			prismaMock.alarm.findUnique.mockResolvedValue(alarm);

			const result = await api.get("/v1/alarms/" + alarm.id);
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should return all Alarms", async () => {
			prismaMock.alarm.findMany.mockResolvedValue(alarms as ResolvedValue<unknown>);

			const result = await api.get("/v1/alarms/");
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should return Alarms filtered by id (through request query params)", async () => {
			const filterIds = alarms.map(a => a.id).slice(0, 2);
			prismaMock.alarm.findMany.mockResolvedValue(alarms.filter(a => filterIds.includes(a.id)) as ResolvedValue<unknown>);

			const result = await api.get("/v1/alarms/?filter_ids=" + filterIds.join(","));
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should return Alarms filtered by id (through request body)", async () => {
			const filterIds = alarms.map(a => a.id).slice(0, 2);
			prismaMock.alarm.findMany.mockResolvedValue(alarms.filter(a => filterIds.includes(a.id)) as ResolvedValue<unknown>);

			const result = await api.get("/v1/alarms/").send({
				filter_ids: filterIds,
			});
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should return only active Alarms", async () => {
			prismaMock.alarm.findMany.mockResolvedValue(alarms.filter(a => a.isActive) as ResolvedValue<unknown>);

			// Take the first two
			const result = await api.get("/v1/alarms/?filter_active=true");
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should return only inactive Alarms", async () => {
			prismaMock.alarm.findMany.mockResolvedValue(alarms.filter(a => !a.isActive) as ResolvedValue<unknown>);

			// Take the first two
			const result = await api.get("/v1/alarms/?filter_active=false");
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should update the Alarm", async () => {
			const newAlarm: Omit<Alarm, "id"> = {
				afspraakId: 999,
				bedrag: 999,
				bedragMargin: 99,
				byDay: [DayOfWeek.Wednesday],
				byMonth: [],
				byMonthDay: [],
				startDate: "2033-12-31",
				endDate: "",
				datumMargin: 10,
				isActive: false,
				signaalId: "71d65065-2eb9-4f44-8f16-44d1e5a4d6f7",
			};

			prismaMock.alarm.update.mockResolvedValue({
				...newAlarm,
				id: alarms[0].id,
			} as ResolvedValue<unknown>);

			const result = await api.put("/v1/alarms/" + alarms[0].id).send(newAlarm);
			expect(result.statusCode).toBe(200);
			expect(result.body).toMatchSnapshot();
		});

		it("should delete the Alarm", async () => {
			prismaMock.alarm.delete.mockResolvedValue(undefined!);

			const result = await api.delete("/v1/alarms/" + alarms[0].id);
			expect(result.statusCode).toBe(204);
		});
	});

	describe("Given the Alarm does not exist", () => {
		it("should return 404 Not Found when tyring to getOneAlarm", async () => {
			prismaMock.alarm.update.mockRejectedValue(new Prisma.PrismaClientKnownRequestError("Not found", "P2025", "Client"));

			const result = await api.get("/v1/alarms/" + alarms[0].id);
			expect(result.statusCode).toEqual(404);
		});

		it("should create the Alarm", async () => {
			const {id, ...alarm} = toAlarm(alarms[0]);
			prismaMock.alarm.create.mockResolvedValue({...alarm, id});

			const result = await api.post("/v1/alarms").send(alarm);
			expect(result.statusCode).toBe(201);
			expect(result.body).toMatchSnapshot();
		});

		it("should return 404 Not Found when trying to updateAlarm", async () => {
			const newStreet = "Barstreet";
			prismaMock.alarm.update.mockRejectedValue(new Prisma.PrismaClientKnownRequestError("Not found", "P2025", "Client"));

			const result = await api.put("/v1/alarms/" + alarms[0].id).send({
				id: alarms[0].id,
				street: newStreet,
			});
			expect(result.statusCode).toEqual(404);
		});

		it("should delete the Alarm", async () => {
			prismaMock.alarm.delete.mockRejectedValue(new Prisma.PrismaClientKnownRequestError("Not found", "P2025", "Client"));
			const result = await api.delete("/v1/alarms/" + alarms[0].id);
			expect(result.statusCode).toEqual(204);
		});
	});

	describe("Given an unknown error occurs on the server", () => {
		const error = new Error("Server did a boo boo");

		it("should return a 500 when tyring to getOneAlarm", async () => {
			prismaMock.alarm.findUnique.mockRejectedValue(error);
			const getAlarmResult = getOneAlarm(alarms[0].id);
			await expect(getAlarmResult).rejects.toEqual(error);

			const result = await api.get("/v1/alarms/" + alarms[0].id);
			expect(result.statusCode).toEqual(500);
		});

		it("should return a 500 when tyring to getManyalarms", async () => {
			prismaMock.alarm.findMany.mockRejectedValue(error);
			const ids = alarms.map(a => a.id);

			const result = await api.get("/v1/alarms/?filter_ids=" + ids.join(","));
			expect(result.statusCode).toEqual(500);
		});

		it("should return a 500 when trying to create the Alarm", async () => {
			prismaMock.alarm.create.mockRejectedValue(error);

			const result = await api.post("/v1/alarms").send(alarms[0]);
			expect(result.statusCode).toBe(500);
			expect(result.body).toMatchSnapshot();
		});

		it("should return a 500 when trying to updateAlarm", async () => {
			prismaMock.alarm.update.mockRejectedValue(error);

			const newStreet = "Barstreet";
			const result = await api.put("/v1/alarms/" + alarms[0].id).send({
				id: alarms[0].id,
				street: newStreet,
			});
			expect(result.statusCode).toEqual(500);
		});

		it("should return a 500 when trying to deleteAlarm", async () => {
			prismaMock.alarm.delete.mockRejectedValue(error);
			const result = await api.delete("/v1/alarms/" + alarms[0].id);
			expect(result.statusCode).toEqual(500);
		});
	});

	describe("Given the server is healthy", () => {
		it("should return 200 OK", async () => {
			const result = await api.get("/health");
			expect(result.statusCode).toEqual(200);
		});
	});

});
