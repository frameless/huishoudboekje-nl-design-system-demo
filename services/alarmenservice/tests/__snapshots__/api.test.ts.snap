// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alarm CRUD (operations) Given an unknown error occurs on the server should return a 500 when trying to create the Alarm 1`] = `
{
  "message": "Server did a boo boo",
  "ok": false,
}
`;

exports[`Alarm CRUD (operations) Given the Alarm does not exist should create the Alarm 1`] = `
{
  "data": {
    "afspraakId": 1,
    "afspraakUuid": "",
    "bedrag": 955,
    "bedragMargin": 75,
    "byDay": [],
    "byMonth": [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
    ],
    "byMonthDay": [
      3,
    ],
    "datumMargin": 7,
    "id": "d9c00acb-8d04-4cf0-9be4-061f33b3357d",
    "isActive": false,
    "startDate": "2021-12-18",
  },
  "ok": true,
}
`;

exports[`Alarm CRUD (operations) Given the Alarm exists should return Alarms filtered by id (through request body) 1`] = `
{
  "data": [
    {
      "afspraakId": 1,
      "afspraakUuid": "",
      "bedrag": 955,
      "bedragMargin": 75,
      "byDay": [],
      "byMonth": [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
      ],
      "byMonthDay": [
        3,
      ],
      "datumMargin": 7,
      "id": "d9c00acb-8d04-4cf0-9be4-061f33b3357d",
      "isActive": false,
      "startDate": "2021-12-18",
    },
    {
      "afspraakId": 2,
      "afspraakUuid": "",
      "bedrag": 299.99,
      "bedragMargin": 15.9,
      "byDay": [],
      "byMonth": [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
      ],
      "byMonthDay": [
        25,
      ],
      "datumMargin": 14,
      "id": "5d65d8e9-1f4a-4cb9-a89c-0f952ac5f2ff",
      "isActive": true,
      "startDate": "2021-12-18",
    },
  ],
  "ok": true,
}
`;

exports[`Alarm CRUD (operations) Given the Alarm exists should return Alarms filtered by id (through request query params) 1`] = `
{
  "data": [
    {
      "afspraakId": 1,
      "afspraakUuid": "",
      "bedrag": 955,
      "bedragMargin": 75,
      "byDay": [],
      "byMonth": [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
      ],
      "byMonthDay": [
        3,
      ],
      "datumMargin": 7,
      "id": "d9c00acb-8d04-4cf0-9be4-061f33b3357d",
      "isActive": false,
      "startDate": "2021-12-18",
    },
    {
      "afspraakId": 2,
      "afspraakUuid": "",
      "bedrag": 299.99,
      "bedragMargin": 15.9,
      "byDay": [],
      "byMonth": [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
      ],
      "byMonthDay": [
        25,
      ],
      "datumMargin": 14,
      "id": "5d65d8e9-1f4a-4cb9-a89c-0f952ac5f2ff",
      "isActive": true,
      "startDate": "2021-12-18",
    },
  ],
  "ok": true,
}
`;

exports[`Alarm CRUD (operations) Given the Alarm exists should return all Alarms 1`] = `
{
  "data": [
    {
      "afspraakId": 1,
      "afspraakUuid": "",
      "bedrag": 955,
      "bedragMargin": 75,
      "byDay": [],
      "byMonth": [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
      ],
      "byMonthDay": [
        3,
      ],
      "datumMargin": 7,
      "id": "d9c00acb-8d04-4cf0-9be4-061f33b3357d",
      "isActive": false,
      "startDate": "2021-12-18",
    },
    {
      "afspraakId": 2,
      "afspraakUuid": "",
      "bedrag": 299.99,
      "bedragMargin": 15.9,
      "byDay": [],
      "byMonth": [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
      ],
      "byMonthDay": [
        25,
      ],
      "datumMargin": 14,
      "id": "5d65d8e9-1f4a-4cb9-a89c-0f952ac5f2ff",
      "isActive": true,
      "startDate": "2021-12-18",
    },
    {
      "afspraakId": 44,
      "afspraakUuid": "",
      "bedrag": 32.05,
      "bedragMargin": 12,
      "byDay": [
        "Monday",
      ],
      "byMonth": [],
      "byMonthDay": [],
      "datumMargin": 1,
      "id": "1a8ac60f-081f-4bbb-baca-6b6d399ce8db",
      "isActive": false,
      "startDate": "2021-12-18",
    },
    {
      "afspraakId": 3,
      "afspraakUuid": "",
      "bedrag": 1200.12,
      "bedragMargin": 75,
      "byDay": [],
      "byMonth": [
        1,
        4,
        7,
        10,
      ],
      "byMonthDay": [
        10,
      ],
      "datumMargin": 3,
      "id": "738247bc-e6b8-41ee-a09e-a6778822598d",
      "isActive": true,
      "startDate": "2021-12-18",
    },
  ],
  "ok": true,
}
`;

exports[`Alarm CRUD (operations) Given the Alarm exists should return only active Alarms 1`] = `
{
  "data": [
    {
      "afspraakId": 2,
      "afspraakUuid": "",
      "bedrag": 299.99,
      "bedragMargin": 15.9,
      "byDay": [],
      "byMonth": [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
      ],
      "byMonthDay": [
        25,
      ],
      "datumMargin": 14,
      "id": "5d65d8e9-1f4a-4cb9-a89c-0f952ac5f2ff",
      "isActive": true,
      "startDate": "2021-12-18",
    },
    {
      "afspraakId": 3,
      "afspraakUuid": "",
      "bedrag": 1200.12,
      "bedragMargin": 75,
      "byDay": [],
      "byMonth": [
        1,
        4,
        7,
        10,
      ],
      "byMonthDay": [
        10,
      ],
      "datumMargin": 3,
      "id": "738247bc-e6b8-41ee-a09e-a6778822598d",
      "isActive": true,
      "startDate": "2021-12-18",
    },
  ],
  "ok": true,
}
`;

exports[`Alarm CRUD (operations) Given the Alarm exists should return only inactive Alarms 1`] = `
{
  "data": [
    {
      "afspraakId": 1,
      "afspraakUuid": "",
      "bedrag": 955,
      "bedragMargin": 75,
      "byDay": [],
      "byMonth": [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
      ],
      "byMonthDay": [
        3,
      ],
      "datumMargin": 7,
      "id": "d9c00acb-8d04-4cf0-9be4-061f33b3357d",
      "isActive": false,
      "startDate": "2021-12-18",
    },
    {
      "afspraakId": 44,
      "afspraakUuid": "",
      "bedrag": 32.05,
      "bedragMargin": 12,
      "byDay": [
        "Monday",
      ],
      "byMonth": [],
      "byMonthDay": [],
      "datumMargin": 1,
      "id": "1a8ac60f-081f-4bbb-baca-6b6d399ce8db",
      "isActive": false,
      "startDate": "2021-12-18",
    },
  ],
  "ok": true,
}
`;

exports[`Alarm CRUD (operations) Given the Alarm exists should return the Alarm 1`] = `
{
  "data": {
    "afspraakId": 1,
    "afspraakUuid": "",
    "bedrag": 955,
    "bedragMargin": 75,
    "byDay": [],
    "byMonth": [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
    ],
    "byMonthDay": [
      3,
    ],
    "datumMargin": 7,
    "id": "d9c00acb-8d04-4cf0-9be4-061f33b3357d",
    "isActive": false,
    "startDate": "2021-12-18",
  },
  "ok": true,
}
`;

exports[`Alarm CRUD (operations) Given the Alarm exists should update the Alarm 1`] = `
{
  "data": {
    "afspraakId": 999,
    "afspraakUuid": "",
    "bedrag": 999,
    "bedragMargin": 99,
    "byDay": [
      "Wednesday",
    ],
    "byMonth": [],
    "byMonthDay": [],
    "datumMargin": 10,
    "endDate": "",
    "id": "d9c00acb-8d04-4cf0-9be4-061f33b3357d",
    "isActive": false,
    "signaalId": "71d65065-2eb9-4f44-8f16-44d1e5a4d6f7",
    "signaalUuid": "",
    "startDate": "2033-12-31",
  },
  "ok": true,
}
`;
